src/util.ts

```diff
@@ -1,3 +1,26 @@
-export const withoutLeadingSlash = (string: string) => string.startsWith('/') ? string.substring(1) : string;
 export const withoutTrailingSlash = (string: string) => string.endsWith('/') ? string.substring(0, string.length - 1) : string;
-export const withTrailingSlash = (string: string) => string.endsWith('/') ? string : (string + '/');
\ No newline at end of file
+export const withTrailingSlash = (string: string) => string.endsWith('/') ? string : (string + '/');
+
+const oldRegions: Array<string> = [
+    "ap-northeast-1",
+    "ap-southeast-1",
+    "ap-southeast-2",
+    "eu-west-1",
+    "sa-east-1",
+    "us-east-1",
+    "us-gov-west-1",
+    "us-west-1",
+    "us-west-2",
+]
+
+// Inspired by Terraform implementation:
+// https://github.com/terraform-providers/terraform-provider-aws/blob/e18168ba0dfd860bbd8f333a93791a2a7eab41db/aws/resource_aws_s3_bucket.go#L1584-L1613
+export const getS3WebsiteDomainUrl = (region: string): string => {
+        // New regions uses different syntax for website endpoints
+        // http://docs.aws.amazon.com/AmazonS3/latest/dev/WebsiteEndpoints.html
+        if (oldRegions.indexOf(region) !== -1) {
```

Comment:
You can use `oldRegions.includes(region)` here, it's more clear than indexOf. [Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes)

Reply:
@JoshuaWalsh I actually wanted to use `.includes` but from what I found when submitting this PR, `Array.includes()` was introduced in ES6. This project seems to be targeting ES5 from what I can see in [`tsconfig.json`](https://github.com/jariz/gatsby-plugin-s3/blob/2e2cffcd8f0db6591805dbfa67c9360940d2e3b0/tsconfig.json#L5) : `"target": "es5",`

Did I miss something? Because it was failing to compile when using `.includes()`

Reply:
Usually not coding in JS, sorry if I'm assuming dumb things :sweat_smile:

Reply:
Oh of course, I forgot #60. `indexOf` is good then.